Source/Mock.cs

```diff
@@ -148,18 +148,56 @@ public static void VerifyAll(params Mock[] mocks)
 		public virtual bool CallBase { get; set; }
 
 		/// <include file='Mock.xdoc' path='docs/doc[@for="Mock.DefaultValue"]/*'/>
+		[Obsolete("Replaced with DefaultValueProvider.")]
 		public virtual DefaultValue DefaultValue
 		{
 			get { return this.defaultValue; }
 			set { this.SetDefaultValue(value); }
 		}
 
+		/// <summary>
+		/// Specifies the class that will determine the default 
+		/// value to return when invocations are made that 
+		/// have no setups and need to return a default 
+		/// value (for loose mocks).
+		/// </summary>
+		public IDefaultValueProvider DefaultValueProvider
+		{
+			get { return this.defaultValueProvider; }
+			set { this.SetDefaultValueProvider(value); }
+		}
+
+		private void SetDefaultValueProvider(IDefaultValueProvider value)
+		{
+			Guard.NotNull(() => value, value);
+
+			if (value.GetType() == typeof(EmptyDefaultValueProvider))
+			{
+				this.defaultValue = DefaultValue.Empty;
+			}
+			else if (value.GetType() == typeof(MockDefaultValueProvider))
+			{
+				this.defaultValue = DefaultValue.Mock;
+			}
+			else
+			{
+				this.defaultValue = DefaultValue.Custom;
+			}
+
+			this.defaultValueProvider = value;
+		}
+
 		private void SetDefaultValue(DefaultValue value)
 		{
+			if (value == DefaultValue.Custom)
+			{
+				throw new ArgumentException("It's impossible that the Custom value is set directly. If you want to use custom default value provider, set the DefaultValueProvider property instead.");
+			}
+
 			this.defaultValue = value;
 			this.defaultValueProvider = defaultValue == DefaultValue.Mock
-				                            ? new MockDefaultValueProvider(this)
-				                            : new EmptyDefaultValueProvider();
+				? (IDefaultValueProvider)new MockDefaultValueProvider(this)
+				: new EmptyDefaultValueProvider();
```

Comment:
Same here. Having both `DefaultValue` and `IDefaultValueProvider` is obviously somewhat redundant (and it's good that the former has been marked as `[Obsolete]` and would probably be phased out), but I wonder if there isn't a more robust approach for "converting" between the two types as long as we have them?